README.md

```diff
@@ -35,32 +40,39 @@ Steps:
 
 - **Setup** Iris: Provide to iris the basic information for it to work by using the `setup(options: LibOptions)`.  
 
-`export interface LibOptions {url:string; exchange: string;}`                                                                                                                                                                                                                                                                                                                                                                                                                                         
+`export interface LibOptions {url:string; exchange: string;}`  
 `url`: Path to RabbitMQ.  
 `exchange`: Name of the exchange queue.  
 
 It would return the functions `{act, add}`
 
-- **Add** a service as provider using `add(pattern: string, implementation: (msg: Buffer) => Promise<Buffer>, opts: AddOptions = {})`:  
-`pattern`: Pattern to which this service will response.  
-`implementation`: Logic to manage the message to return a proper response.  
-`opts`:  `{queue_namespace?: string;}`                             
+- **Add** an implementation to response to a pattern:  
+`add(pattern, implementation, namespace):AddOptions<M, R>`:  
+  - `pattern`: Pattern to which this service will response.  
+  - `implementation`: Logic to manage the message to return a proper response.
+  - `namespace`: Allows to provide several add implementations for the same pattern with out conflict. That way something acting on that pattern will get the responses of all of the implementations an not just one.  
 
 It will return  `Promise<void>`.
 
-- **Get** responses from services  using `act(pattern: string, opts: ActOptions = {}): `  
-`pattern`: Pattern to which this service will response.    
-`opts`:  `{sync?:boolean, timeout?: number, multi?: boolean}`  
+- **Act** on a pattern to get return from the implementations of all the services that had added one to that pattern for an specific payload:  
+`act(pattern: string, payload, timeout?): ActOptions<M> `  
+  - `pattern`: Pattern to which this service will act on.    
+  - `payload`: Message to send as input to the implementation.
+  - `Timeout`: Time after which the wait for answers stops, so all(if any) of the responses gotten till then are receive.  
 
-It will return the proper implementation to get the response needed`((payload: Buffer) => Promise<Buffer>) `
+It will return the proper implementation to get the response needed`Promise<R>`
```

Comment:
This is no longer correct.

Returns a Promise<R> where R is the return type of the remote implementation.